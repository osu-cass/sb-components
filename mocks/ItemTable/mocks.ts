import { ItemTableProps, ItemTableContainerProps, ItemCardModel } from "src/index";
import { AboutItemMockNoRubric } from "mocks/AboutItem/mocks";

export const itemTableProps: ItemTableContainerProps = {
  onRowSelection: () => {},
  onItemSelection: () => {},
  itemCards: [
    {
      selected: false,
      bankKey: 187,
      itemKey: 3206,
      title: "Math Grade 6 Claim 1",
      grade: 8,
      gradeLabel: "Grade 6",
      subjectCode: "MATH",
      subjectLabel: "Math",
      claimCode: "MATH1",
      claimLabel: "Concepts and Procedures",
      targetHash: 100,
      targetId: "A",
      targetShortName: "",
      interactionTypeCode: "EQ",
      interactionTypeLabel: "Equation",
      isPerformanceItem: true,
      brailleOnlyItem: false
    },
    {
      selected: false,
      bankKey: 187,
      itemKey: 3163,
      title: "ELA/ literacy Grade 3 Claim 1",
      grade: 1,
      gradeLabel: "Grade 3",
      subjectCode: "ELA",
      subjectLabel: "ELA/literacy",
      claimCode: "ELA1",
      claimLabel: "Reading",
      targetHash: 1039,
      targetId: "B",
      targetShortName: "Key Details",
      interactionTypeCode: "MC",
      interactionTypeLabel: "Multiple Choice",
      isPerformanceItem: true,
      brailleOnlyItem: false
    },
    {
      selected: false,
      bankKey: 187,
      itemKey: 3615,
      title: "Math Grade 6 Claim 4",
      grade: 8,
      gradeLabel: "Grade 6",
      subjectCode: "MATH",
      subjectLabel: "Math",
      claimCode: "MATH4",
      claimLabel: "Modeling/Data Analysis",
      targetHash: 0,
      targetId: "C",
      targetShortName: "",
      interactionTypeCode: "MS",
      interactionTypeLabel: "Multi Select",
      isPerformanceItem: false,
      brailleOnlyItem: false
    },
    {
      selected: false,
      bankKey: 187,
      itemKey: 2928,
      title: "ELA/ literacy Grade 4 Claim 4",
      grade: 2,
      gradeLabel: "Grade 4",
      subjectCode: "ELA",
      subjectLabel: "ELA/literacy",
      claimCode: "ELA4",
      claimLabel: "Research/Inquiry",
      targetHash: 2832,
      targetId: "D",
      targetShortName: "Evaluate Information/ sources",
      interactionTypeCode: "MS",
      interactionTypeLabel: "Multi Select",
      isPerformanceItem: false,
      brailleOnlyItem: false
    }
  ],
  item: {
    kind: "none"
  }
};

export const itemTableSortProps: ItemTableContainerProps = {
  onRowSelection: () => {},
  onItemSelection: () => {},
  itemCards: [
    {
      selected: false,
      bankKey: 187,
      itemKey: 1,
      title: "alpha",
      grade: 5,
      gradeLabel: "Grade 5",
      subjectCode: "MATH",
      subjectLabel: "Math",
      claimCode: "MATH1",
      claimLabel: "Alpha",
      targetHash: 1,
      targetId: "A",
      targetShortName: "",
      interactionTypeCode: "EQ",
      interactionTypeLabel: "Equation",
      isPerformanceItem: true,
      brailleOnlyItem: false
    },
    {
      selected: false,
      bankKey: 187,
      itemKey: 2,
      title: "bravo",
      grade: 4,
      gradeLabel: "Grade 4",
      subjectCode: "ELA",
      subjectLabel: "ELA/literacy",
      claimCode: "ELA1",
      claimLabel: "Bravo",
      targetHash: 2,
      targetId: "B",
      targetShortName: "",
      interactionTypeCode: "MC",
      interactionTypeLabel: "Multiple Choice",
      isPerformanceItem: true,
      brailleOnlyItem: false
    },
    {
      selected: false,
      bankKey: 187,
      itemKey: 3,
      title: "charlie",
      grade: 3,
      gradeLabel: "Grade 3",
      subjectCode: "MATH",
      subjectLabel: "Math",
      claimCode: "MATH4",
      claimLabel: "Charlie",
      targetHash: 3,
      targetId: "C",
      targetShortName: "",
      interactionTypeCode: "MS",
      interactionTypeLabel: "Multi Select",
      isPerformanceItem: false,
      brailleOnlyItem: false
    },
    {
      selected: false,
      bankKey: 187,
      itemKey: 4,
      title: "delta",
      grade: 2,
      gradeLabel: "Grade 2",
      subjectCode: "ELA",
      subjectLabel: "ELA/literacy",
      claimCode: "ELA4",
      claimLabel: "Delta",
      targetHash: 4,
      targetId: "D",
      targetShortName: "",
      interactionTypeCode: "MS",
      interactionTypeLabel: "Multi Select",
      isPerformanceItem: false,
      brailleOnlyItem: false
    }
  ],
  item: {
    kind: "none"
  }
};

export const itemTableSelectProps: ItemTableContainerProps = {
  onRowSelection: () => {},
  onItemSelection: () => {},
  itemCards: [
    {
      selected: false,
      bankKey: 187,
      itemKey: 1,
      title: "alpha",
      grade: 5,
      gradeLabel: "Grade 5",
      subjectCode: "MATH",
      subjectLabel: "Math",
      claimCode: "MATH1",
      claimLabel: "Alpha",
      targetHash: 1,
      targetId: "A",
      targetShortName: "",
      interactionTypeCode: "EQ",
      interactionTypeLabel: "Equation",
      isPerformanceItem: true,
      brailleOnlyItem: false
    },
    {
      selected: false,
      bankKey: 187,
      itemKey: 2,
      title: "bravo",
      grade: 4,
      gradeLabel: "Grade 4",
      subjectCode: "ELA",
      subjectLabel: "ELA/literacy",
      claimCode: "ELA1",
      claimLabel: "Bravo",
      targetHash: 2,
      targetId: "B",
      targetShortName: "",
      interactionTypeCode: "MC",
      interactionTypeLabel: "Multiple Choice",
      isPerformanceItem: true,
      brailleOnlyItem: false
    },
    {
      selected: false,
      bankKey: 187,
      itemKey: 3,
      title: "charlie",
      grade: 3,
      gradeLabel: "Grade 3",
      subjectCode: "MATH",
      subjectLabel: "Math",
      claimCode: "MATH4",
      claimLabel: "Charlie",
      targetHash: 3,
      targetId: "C",
      targetShortName: "",
      interactionTypeCode: "MS",
      interactionTypeLabel: "Multi Select",
      isPerformanceItem: false,
      brailleOnlyItem: false
    },
    {
      selected: false,
      bankKey: 187,
      itemKey: 4,
      title: "delta",
      grade: 2,
      gradeLabel: "Grade 2",
      subjectCode: "ELA",
      subjectLabel: "ELA/literacy",
      claimCode: "ELA4",
      claimLabel: "Delta",
      targetHash: 4,
      targetId: "D",
      targetShortName: "",
      interactionTypeCode: "MS",
      interactionTypeLabel: "Multi Select",
      isPerformanceItem: false,
      brailleOnlyItem: false
    }
  ],
  item: {
    kind: "success",
    content: AboutItemMockNoRubric
  }
};
