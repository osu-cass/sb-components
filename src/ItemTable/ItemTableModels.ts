import { ItemCardModel } from "../ItemCard/ItemCardModels";

export type HeaderType =
  | "Item"
  | "Claim/Target"
  | "Subject"
  | "Grade"
  | "Item Type";

export enum SortDirection {
  NoSort = 0,
  Ascending = 1,
  Descending = -1
}

export interface HeaderSortModel {
  col: SortColumnModel;
  direction: SortDirection;
  resetSortCount: number;
}

export interface SortColumnModel {
  header: HeaderType;
  className: string;
  accessor: (label: ItemCardModel) => string | number;
  compare: (a: ItemCardModel, b: ItemCardModel) => number;
}

export const headerColumns: SortColumnModel[] = [
  {
    header: "Item",
    className: "item",
    accessor: label => label.itemKey,
    compare: (a, b) => a.itemKey - b.itemKey
  },
  {
    header: "Claim/Target",
    className: "claimAndTarget",
    accessor: label => label.claimLabel + "/" + label.targetShortName,
    compare: (a, b) => {
      if (a.claimCode < b.claimCode || a.targetShortName < b.targetShortName) {
        return SortDirection.Ascending;
      } else if (
        a.claimCode > b.claimCode ||
        a.targetShortName > b.targetShortName
      ) {
        return SortDirection.Descending;
      } else {
        return SortDirection.NoSort;
      }
    }
  },
  {
    header: "Subject",
    className: "subject",
    accessor: label => label.subjectLabel,
    compare: (a, b) => a.subjectCode.localeCompare(b.subjectCode)
  },
  {
    header: "Grade",
    className: "grade",
    accessor: label => label.gradeLabel,
    compare: (a, b) => a.grade - b.grade
  },
  {
    header: "Item Type",
    className: "interactionType",
    accessor: label => label.interactionTypeLabel,
    compare: (a, b) =>
      a.interactionTypeCode.localeCompare(b.interactionTypeCode)
  }
];
