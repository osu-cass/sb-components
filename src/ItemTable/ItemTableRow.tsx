import * as React from "react";
import {
  ItemCardModel,
  ToolTip,
  SortColumnModel,
  ColumnGroup
} from "@src/index";

export interface ItemTableRowProps {
  rowData: ItemCardModel;
  hasControls: boolean;
  columns: ColumnGroup[];
  isExpanded: boolean;
  onRowExpand: (item: ItemCardModel) => void;
  onRowSelect: (item: ItemCardModel) => void;
}

const unChecked = (
  <i className="fa fa-square-o fa-sm table-icon" aria-hidden="true" />
);
const checked = (
  <i className="fa fa-check-square-o fa-sm table-icon" aria-hidden="true" />
);
const collapse = (
  <i className="fa fa-chevron-right fa-sm table-icon" aria-hidden="true" />
);
const expand = (
  <i className="fa fa-chevron-down fa-sm table-icon" aria-hidden="true" />
);

export class ItemTableRow extends React.Component<ItemTableRowProps, {}> {
  constructor(props: ItemTableRowProps) {
    super(props);
  }

  shouldComponentUpdate(nextProps: ItemTableRowProps, nextState: {}) {
    return (
      this.props.isExpanded !== nextProps.isExpanded ||
      this.props.rowData.selected !== nextProps.rowData.selected
    );
  }

  handleRowClick = (rowData: ItemCardModel) => {
    this.props.onRowExpand(rowData);
  };

  handleKeyUpEnter = (
    e: React.KeyboardEvent<HTMLTableRowElement>,
    rowData: ItemCardModel
  ) => {
    if (e.keyCode === 13) {
      this.props.onRowExpand(rowData);
    }
  };

  handleCheckboxClick = (
    e: React.MouseEvent<HTMLTableDataCellElement>,
    rowData: ItemCardModel
  ) => {
    e.stopPropagation();
    this.props.onRowSelect(rowData);
  };

  handleCheckboxKeyUpEnter = (
    e: React.KeyboardEvent<HTMLTableDataCellElement>,
    rowData: ItemCardModel
  ) => {
    if (e.keyCode === 13) {
      e.stopPropagation();
      this.props.onRowSelect(rowData);
    }
  };

  renderColumnGroup(
    colGroup: ColumnGroup,
    cellData: ItemCardModel
  ): JSX.Element {
    const colValues = colGroup.cols.map(c => this.renderCell(c, cellData));
    const { headerClassName } = colGroup;

    return (
      <td
        key={`${headerClassName}-${cellData.bankKey}-${cellData.itemKey}`}
        className={headerClassName}
      >
        {colValues}
      </td>
    );
  }

  renderCell(col: SortColumnModel, cellData: ItemCardModel): JSX.Element {
    const displayText = col.accessor(cellData);
    let content: JSX.Element | undefined;

    if (col.helpText) {
      content = (
        <ToolTip
          helpText={<p>{col.helpText(cellData)}</p>}
          position="bottom"
          displayIcon={true}
          className="box"
        >
          {displayText}
        </ToolTip>
      );
    } else if (col.className === "item" && !this.props.hasControls) {
      content = (
        <a tabIndex={0} role="link">
          {displayText}
        </a>
      );
    } else if (col.className === "subject") {
      content = (
        <span
          className={`table-subject-highlight ${cellData.subjectCode.toLowerCase()}`}
        >
          {displayText}
        </span>
      );
    } else {
      content = <span>{displayText}</span>;
    }

    return <span key={col.className}>{content}</span>;
  }

  renderControls(): JSX.Element[] | undefined {
    const { rowData, hasControls, isExpanded } = this.props;
    let controls: JSX.Element[] | undefined;
    if (hasControls) {
      controls = [
        <td
          className="item-checkbox"
          key="checkbox-control"
          onClick={e => this.handleCheckboxClick(e, rowData)}
          onKeyUp={e => this.handleCheckboxKeyUpEnter(e, rowData)}
          tabIndex={0}
        >
          {rowData.selected === true ? checked : unChecked}&nbsp;
        </td>,
        <td className="arrow-indicator" tabIndex={0} key="expand-control">
          {isExpanded ? expand : collapse}
        </td>
      ];
    }

    return controls;
  }

  render() {
    const { rowData, isExpanded, columns } = this.props;

    return (
      <tr
        key={`${rowData.bankKey}-${rowData.itemKey}`}
        className={isExpanded ? "selected" : ""}
        onClick={() => this.handleRowClick(rowData)}
        onKeyUp={e => this.handleKeyUpEnter(e, rowData)}
      >
        {this.renderControls()}
        {columns.map(col => this.renderColumnGroup(col, rowData))}
      </tr>
    );
  }
}
