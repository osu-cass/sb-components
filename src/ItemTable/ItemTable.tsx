import * as React from "react";
import { HeaderSortModel, SortColumnModel } from "./ItemTableModels";
import { Resource } from "../ApiModel";
import { AboutItemModel } from "../AboutItem/AboutItemModels";
import { ItemCardModel } from "../ItemCard/ItemCardModels";
import { ItemCardViewer } from "../ItemCard/ItemCardViewer";

export interface ItemTableProps {
  mapRows: ItemCardModel[];
  rowOnClick: (item: ItemCardModel) => void;
  sort: HeaderSortModel[];
  columns: SortColumnModel[];
  selectedRow?: ItemCardModel | null;
  item: Resource<AboutItemModel>;
}

export class ItemTable extends React.Component<ItemTableProps, {}> {
  constructor(props: ItemTableProps) {
    super(props);
  }

  handleRowClick = (rowData: ItemCardModel) => {
    this.props.rowOnClick(rowData);
  };

  renderCell(col: SortColumnModel, cellData: ItemCardModel): JSX.Element {
    return (
      <td key={col.header} className={col.className}>
        <div className={col.className}>{col.accessor(cellData)}</div>
      </td>
    );
  }

  //TODO replace X with a > that specifies that  the table  row can  be expanded
  renderRow(rowData: ItemCardModel, index: number): JSX.Element {
    const collapse = (
      <i
        style={{ color: "gray" }}
        className="fa fa-chevron-right fa-sm"
        aria-hidden="true"
      />
    );
    const expand = (
      <i
        style={{ color: "white" }}
        className="fa fa-chevron-down fa-sm"
        aria-hidden="true"
      />
    );
    const style = {
      display: "block",
      padding: "0px"
    };
    let tab = null;
    let isSelected = false;
    if (this.props.selectedRow) {
      isSelected =
        rowData.itemKey === this.props.selectedRow.itemKey &&
        rowData.bankKey === this.props.selectedRow.bankKey;
    }
    let row = (
      <tr
        key={index}
        className={isSelected ? "selected" : ""}
        onClick={() => {
          this.handleRowClick(rowData);
        }}
      >
        <td>{isSelected ? expand : collapse}</td>
        {this.props.columns.map(col => this.renderCell(col, rowData))}
      </tr>
    );
    if (this.props.item.kind === "success" && isSelected) {
      row = (
        <tr key={index}>
          <td colSpan={6}>
            <table className="item-table mapcomponent-table" style={style}>
              <tbody>
                {row}
                <tr>
                  <td colSpan={6}>
                    <ItemCardViewer item={this.props.item.content} />
                  </td>
                </tr>
              </tbody>
            </table>
          </td>
        </tr>
      );
    }

    return row;
  }

  render() {
    return (
      <tbody>
        {this.props.mapRows.map((rowData, idx) => this.renderRow(rowData, idx))}
      </tbody>
    );
  }
}
