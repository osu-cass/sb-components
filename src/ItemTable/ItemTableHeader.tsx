import * as React from "react";
import { SortColumn, HeaderSort, SortDirection } from "./ItemTableModels";

export interface Props {
  columns: SortColumn[];
  onHeaderClick: (header: SortColumn) => void;
  sorts: HeaderSort[];
}

export interface State {}

const invokeResetSortLimit = 2;

const style = {
  color: "gray"
};

const decendingArrow = (
  <span style={style} className="fa fa-sort-desc" aria-hidden="true" />
);

const acendingArrow = (
  <span style={style} className="fa fa-sort-asc" aria-hidden="true" />
);

const noSort = <span style={style} className="fa fa-sort" aria-hidden="true" />;

export class HeaderTable extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      dirElem: undefined
    };
  }

  headerEventHandler(scol: SortColumn, hcol: HeaderSort | undefined) {
    this.props.onHeaderClick(scol);
  }

  setDirElem(headerSort: HeaderSort | undefined): JSX.Element {
    let dirElem = noSort;
    if (!headerSort) {
      return dirElem;
    } else if (headerSort.direction === SortDirection.Ascending) {
      dirElem = acendingArrow;
    } else if (headerSort.direction === SortDirection.Descending) {
      dirElem = decendingArrow;
    }
    return dirElem;
  }

  renderHeader(col: SortColumn): JSX.Element {
    const headerSort = this.props.sorts.find(
      hs => hs.col.header === col.header
    );
    if (headerSort) {
      this.setDirElem(headerSort);
    }

    return (
      <th
        key={col.header}
        className={col.className}
        onClick={() => this.headerEventHandler(col, headerSort)}
      >
        <div className={col.className}>
          <span>{col.header}</span>
          {this.setDirElem(headerSort)}
        </div>
      </th>
    );
  }

  render() {
    return (
      <thead>
        <tr className="primary">
          <th />
          {this.props.columns.map(col => this.renderHeader(col))}
        </tr>
      </thead>
    );
  }
}
