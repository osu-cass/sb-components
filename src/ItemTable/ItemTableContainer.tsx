import * as React from "react";
import { ItemCardModel } from "../ItemCard/ItemCardModels";
import { AboutItemModel } from "../AboutItem/AboutItemModels";
import { Resource } from "../ApiModel";
import {
  HeaderSort,
  SortColumn,
  SortDirection,
  headerColumns
} from "./ItemTableModels";
import { HeaderTable } from "./ItemTableHeader";
import { ItemTable } from "./ItemTable";

export interface ItemTableContainerProps {
  onRowSelection: (
    item: { itemKey: number; bankKey: number },
    reset: boolean
  ) => void;
  itemCards?: ItemCardModel[];
  item: Resource<AboutItemModel>;
}

export interface ItemTableContainerState {
  sorts: HeaderSort[];
  selectedRow?: ItemCardModel;
}

export class ItemTableContainer extends React.Component<
  ItemTableContainerProps,
  ItemTableContainerState
> {
  private pageHeaderColumns = headerColumns;

  constructor(props: ItemTableContainerProps) {
    super(props);
    this.state = {
      sorts: []
    };
  }

  onClickHeader = (col: SortColumn) => {
    const newSorts = (this.state.sorts || []).slice();
    const headIdx = newSorts.findIndex(hs => hs.col.header === col.header);
    if (headIdx !== -1) {
      const newSort = Object.assign({}, newSorts[headIdx]);
      if (newSort.direction == SortDirection.Ascending) {
        newSort.direction = SortDirection.Descending;
      } else if (newSort.direction == SortDirection.Descending) {
        newSort.direction = SortDirection.NoSort;
      } else {
        newSort.direction = SortDirection.Ascending;
      }
      newSorts[headIdx] = newSort;
    } else {
      const newSort: HeaderSort = {
        col: col,
        direction: SortDirection.Ascending,
        resetSortCount: 0
      };
      newSorts.push(newSort);
    }
    this.setState({ sorts: newSorts });
  };

  onSelectItem = (item: ItemCardModel) => {
    const card = { itemKey: item.itemKey, bankKey: item.bankKey };
    if (item === this.state.selectedRow) {
      this.props.onRowSelection(card, true);
      this.setState({ selectedRow: undefined });
    } else {
      this.props.onRowSelection(card, false);
      this.setState({ selectedRow: item });
    }
  };

  invokeMultiSort(lhs: ItemCardModel, rhs: ItemCardModel): number {
    const sorts = this.state.sorts || [];
    for (const sort of sorts) {
      const diff = sort.col.compare(lhs, rhs) * sort.direction;
      if (diff !== 0) {
        return diff;
      }
    }
    return 0;
  }

  getTableData = (): ItemCardModel[] | undefined => {
    let data = this.props.itemCards;
    if (data != undefined) {
      data =
        this.state.sorts && this.state.sorts.length !== 0
          ? data.sort((lhs, rhs) => this.invokeMultiSort(lhs, rhs))
          : data;
    }
    return data;
  };

  renderTableHeader() {
    return (
      <HeaderTable
        sorts={this.state.sorts}
        onHeaderClick={this.onClickHeader}
        columns={this.pageHeaderColumns}
      />
    );
  }

  renderTable() {
    const itemCards = this.getTableData(); //this returns undefined but in the method it has data. that's w
    let content = (
      <span className="placeholder-text" role="alert">
        No results found for the given search terms.
      </span>
    );
    if (itemCards != undefined) {
      //if no items are returned we want to return a friendly message
      if (itemCards.length !== 0) {
        content = (
          <ItemTable
            mapRows={itemCards}
            rowOnClick={this.onSelectItem}
            sort={this.state.sorts}
            columns={this.pageHeaderColumns}
            selectedRow={this.state.selectedRow}
            item={this.props.item}
          />
        );
      }
    }
    return content;
  }

  render() {
    return (
      <div>
        <table className="item-table mapcomponent-table">
          {this.renderTableHeader()}
          {this.renderTable()}
        </table>
      </div>
    );
  }
}
