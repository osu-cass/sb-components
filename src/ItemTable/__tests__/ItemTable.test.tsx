import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ItemTable } from '../ItemTable';
import * as TestUtils from 'react-dom/test-utils';
import { shallow, mount, render } from 'enzyme';


import {
    ItemCardModel,
    GradeLevels,
    RubricModel,
    AboutItemModel,
    Resource
} from '../../index';

describe("ItemTable", () => {
    const itemCards: ItemCardModel[] = [{
        bankKey: 1,
        itemKey: 3,
        title: "",
        grade: GradeLevels.All,
        gradeLabel: "",
        subjectCode: "",
        subjectLabel: "",
        claimCode: "",
        claimLabel: "",
        targetShortName: "",
        interactionTypeCode: "",
        interactionTypeLabel: "",
        isPerformanceItem: true,
        targetHash: 323
    }]

    const rubrics: RubricModel[] = []

    const content: AboutItemModel = {
        rubrics,
        itemCardViewModel: itemCards[0],
        depthOfKnowledge: "depthOfKnowledge",
        targetDescription: "targetDescription",
        commonCoreStandardsDescription: "commonCoreStandardsDescription",
        educationalDifficulty: "educationalDifficulty",
        evidenceStatement: "evidenceStatement"
    }

    const item: Resource<AboutItemModel> = {
        kind: "success",
        content
    }

    const props = {
        tableRef: jest.fn(),
        mapRows: [],
        rowOnClick: jest.fn(),
        sort: [],
        columns: [],
        selectedRow: {
            bankKey: 1,
            itemKey: 123,
            title: "test_title",
            grade: 1,
            gradeLabel: "1",
            subjectCode: "Math",
            subjectLabel: "MTH",
            claimCode: "claim_1234",
            claimLabel: "claim_math",
            targetShortName: "",
            interactionTypeCode: "type01",
            interactionTypeLabel: "type01label",
            isPerformanceItem: false,
            targetHash: 2323
        },
        item
    }

    it("matches snapshot", () => {
        expect(shallow(<ItemTable {...props} />)).toMatchSnapshot();;
    })
})