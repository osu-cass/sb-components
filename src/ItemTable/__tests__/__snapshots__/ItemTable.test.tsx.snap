// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemTable matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemTable
    columns={Array []}
    item={
      Object {
        "content": Object {
          "commonCoreStandardsDescription": "commonCoreStandardsDescription",
          "depthOfKnowledge": "depthOfKnowledge",
          "educationalDifficulty": "educationalDifficulty",
          "evidenceStatement": "evidenceStatement",
          "itemCardViewModel": Object {
            "bankKey": 1,
            "claimCode": "",
            "claimLabel": "",
            "grade": 1023,
            "gradeLabel": "",
            "interactionTypeCode": "",
            "interactionTypeLabel": "",
            "isPerformanceItem": true,
            "itemKey": 3,
            "subjectCode": "",
            "subjectLabel": "",
            "targetHash": 323,
            "targetShortName": "",
            "title": "",
          },
          "rubrics": Array [],
          "targetDescription": "targetDescription",
        },
        "kind": "success",
      }
    }
    mapRows={Array []}
    rowOnClick={[Function]}
    selectedRow={
      Object {
        "bankKey": 1,
        "claimCode": "claim_1234",
        "claimLabel": "claim_math",
        "grade": 1,
        "gradeLabel": "1",
        "interactionTypeCode": "type01",
        "interactionTypeLabel": "type01label",
        "isPerformanceItem": false,
        "itemKey": 123,
        "subjectCode": "Math",
        "subjectLabel": "MTH",
        "targetHash": 2323,
        "targetShortName": "",
        "title": "test_title",
      }
    }
    sort={Array []}
    tableRef={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "tbody",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "tbody",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
